version: '3.8'

services:
  db:
    image: postgres:15
    volumes:
      - postgres_data:/var/lib/postgresql/data
    environment:
      - POSTGRES_DB=sistema_familiar
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
    ports:
      - "5440:5432"
    networks:
      - sistema-familiar-net
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres -d sistema_familiar"]
      interval: 5s
      timeout: 5s
      retries: 5
    deploy:
      mode: replicated
      replicas: 1
      restart_policy:
        condition: on-failure
        max_attempts: 3
      resources:
        limits:
          cpus: '1'
          memory: 1G

  web:
    build: 
      context: .
      args:
        - VERSION=1.0.0
    image: kbokleber/sistema_familiar:1.0.0
    command: python manage.py runserver 0.0.0.0:8001
    volumes:
      - .:/app
      - static_volume:/app/staticfiles
      - media_volume:/app/media
    ports:
      - "8001:8001"
    environment:
      - DJANGO_SETTINGS_MODULE=maintenance_project.settings_dev
      - DATABASE_URL=postgres://postgres:postgres@db:5432/sistema_familiar
      - DOCKER_CONTAINER=true
      - POSTGRES_DB=sistema_familiar
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
      - POSTGRES_HOST=db
      - POSTGRES_PORT=5432
    depends_on:
      - db
    networks:
      - sistema-familiar-net
    deploy:
      mode: replicated
      replicas: 1
      restart_policy:
        condition: on-failure
        max_attempts: 3
      resources:
        limits:
          cpus: '1'
          memory: 1G

volumes:
  postgres_data:
    driver: local
  static_volume:
    driver: local
  media_volume:
    driver: local

networks:
  sistema-familiar-net:
    driver: overlay
    attachable: true 